### ISSUE TITLE
false negative "too-many-arguments"

### ISSUE THREAD

**Issue by buhtz** [pylint-dev/pylint#9547]
### Bug description

```python
Original Code: https://codeberg.org/buhtz/buhtzology/src/commit/65f71ca1b5621fa3e3400ec0bcdd5de84ec206db/src/buhtzology/stausberg.py#L141-L149

Expect "too-many-arguments" because here are 6 instead of 5 arguments. But pylint do not detect that.


def weight_by_icd_group(data: pandas.DataFrame,
                        index_column: str,
                        icd_column: str,
                        weight_name: str = 'weight',
                        ignored_icds: tuple = ('UUU'),
                        exclude_appendix_artifact: bool = True
                        ) -> pandas.DataFrame:
    pass
```
```


### Configuration

```ini
no config
```

### Command used

```shell
pylint src.py
```

### Pylint output

```python
No output because no error.
```

### Expected behavior

too-many-arguments error

### Pylint version

```shell
pylint 3.0.3                                                                                                                  astroid 3.0.2                                                                                                                 Python 3.9.2 (default, Feb 28 2021, 17:03:44)                                                                                 [GCC 10.2.1 20210110]
```

### OS / Environment

Debian 11

**Comment by mbyrnepr2** [comment]
Hey @buhtz it looks like expected behaviour because one of the parameter names begins with the string [ignored_](https://github.com/pylint-dev/pylint/blob/bb9e0797eeacd8c415869fb70ceef17d27643c58/pylint/checkers/variables.py#L45).
That parameter is [excluded](https://github.com/pylint-dev/pylint/blob/main/pylint/checkers/design_analysis.py#L542) from the count of total parameters.

**Comment by buhtz** [comment]
Technically I do understand. But what is the reason behind it? How does this improve code quality?

**Comment by jacobtylerwalls** [comment]
The reason is that the default value for the `--ignored-argument-names` setting includes `ignored`, which you are using in your snippet. That's unlucky in your case, but it has a fix -- you can set `--ignored-argument-names` to something else if you have a legitimate use case for an argument name that begins with that word.

**Comment by buhtz** [comment]
I am still not convinced but I would minimally treat this as a bug in documentation. The docu page about that error/rule should mention the exceptional cases caused by ignoored-argument-names.

**Comment by Pierre-Sassoulas** [comment]
This would be greatly alleviated by #9184
